src/Compilers/Core/Portable/Compilation/SymbolInfo.cs

```diff
@@ -91,35 +72,20 @@ internal SymbolInfo(ISymbol? symbol, ImmutableArray<ISymbol> candidateSymbols, C
             this.CandidateReason = candidateReason;
         }
 
+        internal ImmutableArray<ISymbol> GetAllSymbols()
+            => this.Symbol == null ? CandidateSymbols : ImmutableArray.Create(this.Symbol);
+
         public override bool Equals(object? obj)
-        {
-            return obj is SymbolInfo && Equals((SymbolInfo)obj);
-        }
+            => obj is SymbolInfo info && Equals(info);
 
         public bool Equals(SymbolInfo other)
-        {
-            if (!object.Equals(this.Symbol, other.Symbol) ||
-                _candidateSymbols.IsDefault != other._candidateSymbols.IsDefault ||
-                this.CandidateReason != other.CandidateReason)
-            {
-                return false;
-            }
-
-            return _candidateSymbols.IsDefault || _candidateSymbols.SequenceEqual(other._candidateSymbols);
-        }
+            => this.CandidateReason == other.CandidateReason &&
+               Equals(this.Symbol, other.Symbol) &&
```

Comment:
What version of `Equals` is this? Please qualify, or use `ReferenceEqual`.

Reply:
will do.